pymatgen/io/cif.py

```diff
@@ -1196,8 +1203,7 @@ def parse_structures(
                 "in the CIF file as is. If you want the primitive cell, please set primitive=True explicitly.",
                 UserWarning,
             )
-        if not check_occu:  # added in https://github.com/materialsproject/pymatgen/pull/2836
-            warnings.warn("Structures with unphysical site occupancies are not compatible with many pymatgen features.")
```

Comment:
This warning should not be raised just because `if not check_occu`, it should really check the occupancy and only warn if the occupancy is "unphysical".